sdk/servicebus/azure-servicebus/tests/servicebus_preparer.py

```diff
@@ -0,0 +1,389 @@
+                                                     disable_recording=disable_recording,
+                                                     playback_fake_resource=playback_fake_resource,
+                                                     client_kwargs=client_kwargs)
+        self.parameter_name = parameter_name
+        self.access_rights = access_rights
+        self.servicebus_queue_parameter_name = servicebus_queue_parameter_name
+
+    def create_resource(self, name, **kwargs):
+        if self.is_live:
+            self.client = self.create_mgmt_client(ServiceBusManagementClient)
+            group = self._get_resource_group(**kwargs)
+            namespace = self._get_namespace(**kwargs)
+            queue = self._get_queue(**kwargs)
+            self.resource = self.client.queues.create_or_update_authorization_rule(
+                group.name,
+                namespace.name,
+                queue.name,
+                name,
+                self.access_rights
+            )
+
+            key = self.client.queues.list_keys(group.name, namespace.name, queue.name, name)
+            connection_string = key.primary_connection_string
+        else:
+            self.resource = FakeResource(name=name, id=name)
+            connection_string = 'https://microsoft.com'
+        return {
+            self.parameter_name: self.resource,
+            '{}_connection_string'.format(self.parameter_name): connection_string,
+        }
+
+    def remove_resource(self, name, **kwargs):
+        if self.is_live:
+            group = self._get_resource_group(**kwargs)
+            namespace = self._get_namespace(**kwargs)
+            queue = self._get_queue(**kwargs)
+            self.client.queues.delete_authorization_rule(group.name, namespace.name, queue.name, name, polling=False)
+
+    def _get_queue(self, **kwargs):
+        try:
+            return kwargs.get(self.servicebus_queue_parameter_name)
+        except KeyError:
+            template = 'To create this service bus queue authorization rule a service bus queue is required. Please add ' \
+                       'decorator @{} in front of this service bus preparer.'
+            raise AzureTestError(template.format(ServiceBusQueuePreparer.__name__))
+
+
+# Old servicebus tests were not written to work on both stubs and live entities.
+# This disables those tests for non-live scenarios, and should be removed as tests
+# are ported to offline-compatible code.
+def AreLiveTestsEnabled():
+    # This is how the CI pipe denotes that it's a live test run.
+    if os.environ.get('AZURE_RUN_MODE', None) == 'Live':
+        return True
+    # This is how local test runs denote that they're live.
+    # Note: This isn't amazing, because it could be present but not populated,
+    # but it's isomorphic to the old skips where it'd only check if the env vars were present.
+    try:
+        from devtools_testutils import mgmt_settings_real
```

Comment:
```python
from devtools_testutils import is_live
return is_live
```
that will be more accurate

Reply:
if your CI also defines AZURE_TEST_RUN_LIVE, you can even also completly skip this method totally and use the live_test_only that will do exactly your skip if for free.
https://github.com/Azure/azure-sdk-for-python/blob/01447005686fab41d243d6bb6834653e1ce1fe33/sdk/storage/azure-storage-blob/tests/test_large_block_blob.py#L79